kytos/core/db.py

```diff
@@ -0,0 +1,139 @@
+from pymongo.errors import AutoReconnect, OperationFailure
+
+from kytos.core.exceptions import KytosDBInitException
+
+LOG = logging.getLogger(__name__)
+
+
+def _log_pymongo_thread_traceback() -> None:
+    """Log pymongo thread traceback.
+
+    Originally, it printed out to sys.stderr poluting it when handling
+    certain asynchronous errors that can happen, with this patched function
+    it logs to LOG.error instead.
+    """
+    if sys.stderr:
+        einfo = sys.exc_info()
+        try:
+            LOG.error(einfo)
+        except IOError:
+            pass
+        finally:
+            del einfo
+
+
+if hasattr(pymongo.helpers, "_handle_exception"):
+    pymongo.helpers._handle_exception = _log_pymongo_thread_traceback
+
+
+def mongo_client(
+    host_seeds=os.environ.get(
+        "MONGO_HOST_SEEDS", "mongo1:27017,mongo2:27018,mongo3:27019"
+    ),
+    username=os.environ.get("MONGO_USERNAME") or "invalid_user",
+    password=os.environ.get("MONGO_PASSWORD") or "invalid_password",
+    database=os.environ.get("MONGO_DBNAME") or "napps",
+    connect=False,
+    retrywrites=True,
+    retryreads=True,
+    readpreference="primaryPreferred",
+    readconcernlevel="majority",
+    maxpoolsize=int(os.environ.get("MONGO_MAX_POOLSIZE") or 300),
+    minpoolsize=int(os.environ.get("MONGO_MIN_POOLSIZE") or 30),
+    serverselectiontimeoutms=30000,
+    **kwargs,
+) -> MongoClient:
+    """Instantiate a MongoClient instance.
+
+    MongoClient is thread-safe and has connection-pooling built in. NApps are
+    supposed to use the Mongo class that wraps a MongoClient. NApps might
+    use a new MongoClient instance for exceptional cases where a NApp
+    needs to parametrize differently.
+    """
+    return MongoClient(
+        host_seeds.split(","),
+        username=username,
+        password=password,
+        connect=connect,
+        authsource=database,
+        retrywrites=retrywrites,
+        retryreads=retryreads,
+        readpreference=readpreference,
+        maxpoolsize=maxpoolsize,
+        minpoolsize=minpoolsize,
+        readconcernlevel=readconcernlevel,
+        serverselectiontimeoutms=serverselectiontimeoutms,
+        **kwargs,
+    )
+
+
+class Mongo:
+    """MongoClient instance for NApps."""
+
+    client = mongo_client(connect=False)
+    db_name = os.environ.get("MONGO_DBNAME") or "napps"
+
+    @classmethod
+    def bootstrap_index(
+        cls,
+        collection: str,
+        index: str,
+        direction: int,
+        **kwargs,
+    ) -> Optional[str]:
+        """Bootstrap index."""
+        db = cls.client[cls.db_name]
+        indexes = set()
+
+        for value in db[collection].index_information().values():
+            if "key" in value and isinstance(value["key"], list):
+                indexes.add(value["key"][0])
+
+        if (index, direction) not in indexes:
+            return db[collection].create_index([(index, direction)], **kwargs)
+
+        return None
```

Comment:
This doesn't support creating compound indexes. I know that is the most likely use case, but shouldn't compound indexes be supported?

Reply:
@ajoaoff indeed, that's a great point, especially since it's the same `create_index` function, so this class method should support for it, while checking if they have already been created, I'll prototype and explore it, let's see how it goes.

Reply:
@ajoaoff done on d845a462a3f815514e4db88b65452d5dfd80d08e, I've also taken the opportunity to set the index creation as `background` by default.